Offroad-iOS/Offroad-iOS/Presentation/QuestList/Views/Components/CourseQuestCollectionViewCell.swift

```diff
@@ -0,0 +1,200 @@
+            stackView.distribution = .equalSpacing
+        }
+    }
+    
+    private func setupLayout() {
+        widthConstraint.priority = .defaultHigh
+        widthConstraint.isActive = true
+        
+        courseQuestNameLabel.snp.makeConstraints { make in
+            make.top.equalToSuperview().inset(18)
+            make.leading.equalToSuperview().inset(20)
+            make.trailing.equalTo(courseQuestProgressLabel.snp.leading).offset(-7)
+        }
+        
+        courseQuestProgressLabel.snp.makeConstraints { make in
+            make.centerY.equalTo(courseQuestNameLabel)
+            make.trailing.equalTo(chevronImageView.snp.leading).offset(-10)
+        }
+        
+        chevronImageView.snp.makeConstraints { make in
+            make.top.equalToSuperview().inset(8)
+            make.trailing.equalToSuperview()
+            make.size.equalTo(44)
+        }
+        
+        courseQuestDescriptionLabel.snp.makeConstraints { make in
+            make.top.equalTo(courseQuestNameLabel.snp.bottom).offset(8)
+            make.horizontalEdges.equalToSuperview().inset(20)
+        }
+        
+        courseQuestListStackView.snp.makeConstraints { make in
+            make.horizontalEdges.equalToSuperview().inset(12)
+            make.verticalEdges.equalToSuperview().inset(9)
+        }
+        
+        courseQuestInfoView.snp.makeConstraints { make in
+            make.top.equalTo(courseQuestDescriptionLabel.snp.bottom).offset(14)
+            make.horizontalEdges.equalToSuperview().inset(12)
+        }
+        
+        expandedBottomConstraint.priority = .defaultLow
+        shrinkedBottomConstraint.priority = .defaultLow
+        
+        expandedBottomConstraint.isActive = isSelected
+        shrinkedBottomConstraint.isActive = !isSelected
+    }
+    
+    private func setAppearance() {
+        expandedBottomConstraint.isActive = isSelected
+        shrinkedBottomConstraint.isActive = !isSelected
+        
+        courseQuestDescriptionLabel.isHidden = !isSelected
+        courseQuestInfoView.isHidden = !isSelected
+        
+        let rotationTransform = isSelected ? CGAffineTransform(rotationAngle: .pi * 0.999) : CGAffineTransform.identity
+        chevronImageView.transform = rotationTransform
+        contentView.layoutIfNeeded()
+    }
+    
+    // MARK: - Configure Cell
+    
+    func configureCell(with quest: CourseQuest) {
+        courseQuestNameLabel.text = quest.title
+        courseQuestProgressLabel.text = "달성도 (\(quest.progress))"
+        courseQuestProgressLabel.highlightText(targetText: "달성도", color: .grayscale(.gray400))
+        courseQuestDescriptionLabel.text = quest.description
+        courseQuestListStackView.arrangedSubviews.forEach { $0.removeFromSuperview() }
+        
+        for questDetail in quest.quests {
+            let questStackView = IconLabelStackView(icon: .icnQuestListCheckBox, text: "\(questDetail.locationName): \(questDetail.mission)")
```

Comment:
P3)
`IconLabelStackView`를 초기화하면서 생성자의 인자로 `UIImage` 타입의 인스턴스인 `.icnQuestListCheckBox` 가 전달됩니다. 그런데 지금은 여러 IconLabelStackView를 여러 번 생성하면서 같은 이미지를 여러 번 생성합니다.
만약 지금과 같이 같은 이미지를 반복해서 사용하는 경우라면, .icnQuestListCheckBox 를 하나만 만들고 생성자에 UIImage 를 전달할 때 이를 전달하여 하나의 UIImage 인스턴스를 공유하는 것이 성능상 이점이 있습니다. 
(불필요한 Heap allocation은 성능에 부정적인 영향을 미칩니다.)
```suggestion
        let checkBoxImage: UIImage = .icnQuestListCheckBox
        for questDetail in quest.quests {
            let questStackView = IconLabelStackView(icon: checkBoxImage, text: "\(questDetail.locationName): \(questDetail.mission)")
```
만약 다양한 종류의 이미지를 사용한다면 이러한 방법은 무의미해질 수 있으므로, 실제 디자인과 서버 데이터가 어떤 식으로 구현되는지에 따라 달라질 수 있습니다.

Reply:
좋은 지적 감사합니다! 말씀해주신 대로 같은 이미지를 반복 생성하는 부분은 메모리 효율과 성능 측면에서 개선해야 할 것 같습니다.
이미지 인스턴스를 사전에 생성해두고 루프 내에서 재사용하는 방식으로 변경하겠습니다.